src/plugins/advanced_settings/public/management_app/components/form/form.tsx

```diff
@@ -163,6 +165,11 @@ export class Form extends PureComponent<FormProps> {
           const isArray = Array.isArray(JSON.parse((defVal as string) || '{}'));
           valueToSave = valueToSave.trim();
           valueToSave = valueToSave || (isArray ? '[]' : '{}');
+        case 'boolean':
+          if (metric && this.props.trackUiMetric) {
+            const metricName = valueToSave ? `${metric.name}_on` : `${metric.name}_off`;
+            this.props.trackUiMetric(metric.type, metricName);
+          }
```

Comment:
We are counting the number of ON/OFF events, but we don't know the final value. Is this the intention?

For the final selected value, `stack_stats.kibana.plugins.stack_management` already reports it when it's different to the default. Would that work?

Reply:
Not sure what you mean by "we don't know the final value"? `valueToSave` is the final value that is going to be saved.

> For the final selected value, stack_stats.kibana.plugins.stack_management already reports it when it's different to the default. Would that work?

This is great insight, thanks. I'm not sure how it works, will have a look at it. Would it report if a value changes from non-default to a default one?

Reply:
> Not sure what you mean by "we don't know the final value"? `valueToSave` is the final value that is going to be saved.

If we report:

```
{
  "ui_metric": {
    "modifyColumns_on": 1,
    "modifyColumns_off": 1
  }
}
```

Do we assume it's ON because it's ON by default but it went OFF once and then ON again?


> > For the final selected value, stack_stats.kibana.plugins.stack_management already reports it when it's different to the default. Would that work?
> 
> This is great insight, thanks. I'm not sure how it works, will have a look at it. Would it report if a value changes from non-default back to a default one?

FYI: this the implementation: https://github.com/elastic/kibana/blob/18f7f042c1b6bd36c0cf09c9fed4396e7484e0a4/src/plugins/kibana_usage_collection/server/collectors/management/telemetry_management_collector.ts#L49

If the default value is ON, it will report as follows:

| User's option | Reported `stack_stats.kibana.plugins.stack_management.modifyColumns` |
|---------------|--------------------|
| ON (same as default) | the key won't be reported |
| OFF | `false` |
| ON again | the key won't be reported again because it matches the default |

Reply:
> Do we assume it's ON because it's ON by default but it went OFF once and then ON again?

Yes, it means it's on. I think just a rough sense of the ratio of how many times users have switched on/off is a good metric. If it was switched off 50 times and switched back on 0 times - users must be having an issue with the default implementation. If it was switched off 50 times and switched back on 45 times - majority of users have tried the default behavior, compared it with the old behavior, and decided they like the new behavior better. I think the ratio here is a valuable metric and without reporting `ON again` scenario, we'd lose that insight.

Reply:
Sounds good to me.
@alexfrancoeur